src/connector/postgres/error.rs

```diff
@@ -44,7 +54,12 @@ impl From<tokio_postgres::error::Error> for Error {
                 let splitted: Vec<&str> = splitted[1].split('"').collect();
                 let db_name = splitted[1].into();
 
-                Error::DatabaseDoesNotExist { db_name }
+                let mut builder = Error::builder(ErrorKind::DatabaseDoesNotExist { db_name });
+
+                builder.set_original_code("3D000");
```

Comment:
it worries me a tiny bit that the error code is repeated here from the pattern, and they could get out of sync. It's less pretty, but the pattern could be `Some(code) if code = "3D000"`, then here it can be `builder.set_original_code(code)`. Just a suggestion though, I don't think it's too bad how it is.

Reply:
hwut? done.